android/src/com/mapswithme/maps/maplayer/SearchWheel.java

```diff
@@ -100,33 +109,61 @@ public static SearchOption fromSearchQuery(@NonNull String query, @NonNull Conte
     }
   }
 
-  SearchWheel(View frame)
+  public SearchWheel(View frame, @NonNull View.OnClickListener onSearchPressedListener, @NonNull View.OnClickListener onSearchCanceledListener)
   {
     mFrame = frame;
-
+    mOnSearchPressedListener = onSearchPressedListener;
+    mOnSearchCanceledListener = onSearchCanceledListener;
     mTouchInterceptor = mFrame.findViewById(R.id.touch_interceptor);
-    mTouchInterceptor.setOnClickListener(this);
+    if (mTouchInterceptor != null)
+      mTouchInterceptor.setOnClickListener(this);
     mSearchButton = mFrame.findViewById(R.id.btn_search);
     mSearchButton.setOnClickListener(this);
-    mSearchLayout = mFrame.findViewById(R.id.search_frame);
-    if (UiUtils.isLandscape(mFrame.getContext()))
+    refreshSearchVisibility();
+  }
+
+  private @Nullable View getSearchLayout()
+  {
+    if (mSearchLayout == null)
     {
-      UiUtils.waitLayout(mSearchLayout, () -> {
-        mSearchLayout.setPivotX(0);
-        mSearchLayout.setPivotY(mSearchLayout.getMeasuredHeight() / 2);
-      });
+      mSearchLayout = mFrame.findViewById(R.id.search_frame);
+      if (mSearchLayout != null)
+      {
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        WindowManager windowmanager = (WindowManager) mFrame.getContext().getSystemService(Context.WINDOW_SERVICE);
+        windowmanager.getDefaultDisplay().getMetrics(displayMetrics);
+        // Get available screen height in DP
+        int height =  Math.round(displayMetrics.heightPixels / displayMetrics.density);
+        // If height is less than 400dp, the search wheel in a straight line
+        // In this case, move the pivot for the animation
+        if (height < 400)
```

Comment:
nit: It looks unusual for me, but I don't understand nothing in Android layouts :)

Reply:
The search wheel has 2 possible shapes: round or straight line. Before this PR, the round shape was used in portrait mode, and the straight line in landscape. But now this PR uses the available height to decide which shape to show (threshold is 400dp). So instead of checking the orientation, this function must now check the available height to understand which shape is loaded.

I agree this is not the prettiest code but I could not find a better way to handle this.